bmcs/src/main/scala/akka/stream/alpakka/oracle/bmcs/impl/BmcsStream.scala

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.oracle.bmcs.impl
+
+import java.nio.file.Paths
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.marshalling.Marshal
+import akka.http.scaladsl.model.headers.ByteRange
+import akka.http.scaladsl.model.{RequestEntity, _}
+import akka.http.scaladsl.unmarshalling.{Unmarshal, Unmarshaller}
+import akka.stream.alpakka.oracle.bmcs.{BmcsException, BmcsSettings, DiskBufferType, MemoryBufferType}
+import akka.stream.Materializer
+import akka.stream.alpakka.oracle.bmcs.auth.{BmcsCredentials, BmcsSigner}
+import akka.stream.alpakka.oracle.bmcs.scaladsl.ListObjectsResultContents
+import akka.stream.alpakka.s3.impl.{DiskBuffer, MemoryBuffer, SplitAfterSize}
```

Comment:
This isn't great, having one module depend on internals of another, but maybe it's worth it for this module.

Reply:
yes. I agree. let's not repeat ourselves.